### ISSUE TITLE
Again annoying bugs in the new release of Mypal 28.8.4

### ISSUE THREAD

**Issue by eugen1962** [Feodor2/Mypal#154]
1. The service pack installer again creates a section and registry entries:

HKEY_LOCAL_MACHINE \ SOFTWARE \ Mypal 28.8.4 \ extensions

Components REG_SZ C: \ Intel \ Mypal \ components
Plugins    REG_SZ C: \ Intel \ Mypal \ plugins

However, the installer does not create these sections in the Mypal directory.
On the contrary!
The installer deletes the "components" section if you created it yourself earlier (created for a completely understandable reason).
The "plugins" section you created installer does not delete.

This bug, by the way, I previously observed in some builds of Mozilla. I ask developers to fix this annoying mistake.

2. In the Mypal 28.8.4 package, the necessary files are again missing:

xul.dll requires concrt140.dll, api-ms-win-core-winrt-l1-1-0.dll, api-ms-win-core-winrt-string-l1-1-0.dll (+ api-ms-win-core-winrt-error-l1-1-0.dll)

libEGL.dll, libGLESv2.dll, mozglue.dll require concrt140.dll

I was able to run Mypal in WebGL mode (AMD Radeon HD 3450 AGP (RV620 LE AGP) Hightech / Catalyst ver: 14.4 /  OpenGL 3.0+ / GLSL: 3.30) only when:

2.1. Installed

2.1.1. The latest supported Visual C ++ downloads Visual Studio 2015, 2017 and 2019
https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads

2.1.2. DirectX End-User Runtime Web Installer (it has already been installed on my system, including d3dcompiler_47.dll and d3dcsx_46.dll)
https://www.microsoft.com/en-us/download/details.aspx?id=35&nowin10

2.1.3. api-ms-win-core-winrt-l1-1-0.dll
https://www.dll-files.com/api-ms-win-core-winrt-l1-1-0.dll.html

api-ms-win-core-winrt-string-l1-1-0.dll
https://www.dll-files.com/api-ms-win-core-winrt-string-l1-1-0.dll.html

2.1.4. api-ms-win-core-winrt-error-l1-1-0.dll
https://www.dll-files.com/api-ms-win-core-winrt-error-l1-1-0.dll.html

2.2. Deleted the native libraries from the Mypal directory:

api_ms_win_core_console_l1_1_0.dll
api-ms-win-core-console-l1-1-0.dll
api-ms-win-core-datetime-l1-1-0.dll
api-ms-win-core-debug-l1-1-0.dll
api-ms-win-core-errorhandling-l1-1-0.dll
api-ms-win-core-file-l1-1-0.dll
api-ms-win-core-file-l1-2-0.dll
api-ms-win-core-file-l2-1-0.dll
api-ms-win-core-handle-l1-1-0.dll
api-ms-win-core-heap-l1-1-0.dll
api-ms-win-core-interlocked-l1-1-0.dll
api-ms-win-core-libraryloader-l1-1-0.dll
api-ms-win-core-localization-l1-2-0.dll
api-ms-win-core-memory-l1-1-0.dll
api-ms-win-core-namedpipe-l1-1-0.dll
api-ms-win-core-processenvironment-l1-1-0.dll
api-ms-win-core-processthreads-l1-1-0.dll
api-ms-win-core-processthreads-l1-1-1.dll
api-ms-win-core-profile-l1-1-0.dll
api-ms-win-core-rtlsupport-l1-1-0.dll
api-ms-win-core-string-l1-1-0.dll
api-ms-win-core-synch-l1-1-0.dll
api-ms-win-core-synch-l1-2-0.dll
api-ms-win-core-sysinfo-l1-1-0.dll
api-ms-win-core-timezone-l1-1-0.dll
api-ms-win-core-util-l1-1-0.dll
api-ms-win-crt-conio-l1-1-0.dll
api-ms-win-crt-convert-l1-1-0.dll
api-ms-win-crt-environment-l1-1-0.dll
api-ms-win-crt-filesystem-l1-1-0.dll
api-ms-win-crt-heap-l1-1-0.dll
api-ms-win-crt-locale-l1-1-0.dll
api-ms-win-crt-math-l1-1-0.dll
api-ms-win-crt-multibyte-l1-1-0.dll
api-ms-win-crt-private-l1-1-0.dll
api-ms-win-crt-process-l1-1-0.dll
api-ms-win-crt-runtime-l1-1-0.dll
api-ms-win-crt-stdio-l1-1-0.dll
api-ms-win-crt-string-l1-1-0.dll
api-ms-win-crt-time-l1-1-0.dll
api-ms-win-crt-utility-l1-1-0.dll
D3DCompiler_43.dll
d3dcompiler_47.dll
msvcp140.dll
ucrtbase.dll
vcruntime140.dll

2.3. Reduced the dependentlibs.list file:

mozglue.dll
nss3.dll
lgpllibs.dll
xul.dll

3. Optional. It is worth checking the keys in about:config and setting the correct values corresponding to the capabilities of hardware and drivers (GPU_Caps_Viewer etc. to help you). For example

Toggle webgl.angle.force-d3d11 and webgl.angle.try-d3d11 to false (WindowsXP).
Toggle webgl.disabled to true to disable WebGL, or to false to enable WebGL.
To enable WebGL 2 toggle webgl.enable-webgl2 to true, to disable - false.

To see all WebGL extensions in Mypal, browse to about:config and set webgl.enable-debug-renderer-info, webgl.enable-draft-extensions and webgl.enable-privileged-extensions to true, go to...
https://browserleaks.com/webgl
https://webglreport.com/
https://www.geeks3d.com/webgl/
etc... 
After it is necessary (!) to set webgl.enable-debug-renderer-info, webgl.enable-draft-extensions and webgl.enable-privileged-extensions to false.

After correction about:config Mypal works very quickly. This consumes significantly less system resources than using alternative browsers.

========================

Returning to points 1-4... I probably understand why so many files, target packages of which Microsoft distributes and regularly updates (!), Mozilla, Maxthon include in all their distributions, as well as why some of the necessary libraries are missing in their free distributions.
But I don’t understand why it is to such an excellent developer as Feodor2.

Of course, it is not difficult to remove the libraries with each update, but maybe the package should be lightened?
Why spoil such a masterpiece?
However, one could make 2 variants of the distribution: 1 - with a complete set of files in the Mypal directory (Starter), 2 - a lightweight elegant distribution (Professional) for those who already have the necessary packages installed or who are able to install the missing Microsoft files on their own.

Mypal is the best browser for WindowsXP, in my opinion, out of the 3 remaining stable and popular products (Maxthon, Mypal, Otter Browser; others either stopped developing or browsers are unstable). The number of downloads suggests that those who consider WindowsXP to be an outdated operating system are wrong. Many thanks to Feodor2 for such a great development!

**Comment by Feodor2** [comment]
Why again? It was always like this, and comes from the mozilla, why this practice is going is another question. I want to get the actually reason for consider.

>The service pack installer again creates a section and registry entries

Where did you get it, i publish one regular installer only it is not to touch registry in the portable mode.


I can do an option in the installer maybe, if copying dlls is critically annoying for another one.

And i shall consider about webgl settings

**Comment by Feodor2** [comment]
By the way try mingw build #126 without api_ms_win_core.. bunch, even if it has couple issues i use it as the regular browser and keep publish new versions.

**Comment by eugen1962** [comment]
> 
> 
> By the way try mingw build #126 without api_ms_win_core.. bunch, even if it has couple issues i use it as the regular browser and keep publish new versions.

Great, thank you! Version mypal-28.8.3_MinGW_VS2010 (# 126) has already been installed. 

mypal-28.8.3_MinGW_VS2010
This browser scored 406 out of 555 points
This score was recorded using Firefox 68.9 on Windows XP
Useragent: Mozilla / 5.0 (Windows NT 5.1; rv: 68.9) Gecko / 20100101 Goanna / 4.4 Firefox / 68.9 Mypal / 28.8.3
Screen size: 1360 x 768 px
http://html5test.com/s/6eef944b6f04d73a.html

mypal-28.8.4. (full)
This browser scored 406 out of 555 points
This score was recorded using Firefox 68.9 on Windows XP
Useragent: Mozilla / 5.0 (Windows NT 5.1; rv: 68.9) Gecko / 20100101 Goanna / 4.4 Firefox / 68.9 Mypal / 28.8.4
Screen size: 1360 x 768 px
http://html5test.com/s/36429b4b6f08bd94.html

**Comment by eugen1962** [comment]
> 
> 
> By the way try mingw build #126 without api_ms_win_core.. bunch, even if it has couple issues i use it as the regular browser and keep publish new versions.

Of course, the installer bug is inherited from Mozilla.
Of course, it's not about the portable version of Mypal. 
Nothing is critical, I was simply annoyed, and did not express a claim.

About WebGL based on your comment (mypal-28.8.3_MinGW_VS2010). 
I think that the driver driver 8.970.100.0 (07/01/2014) that I found and installed for the ATI Radeon HD 3450 required some registered library from MSVS. Therefore, I did not work with WebGL before installing the package and registering the libraries.
About the availability of WebGL2 and Direct3D11 in the browser code for WindowsXP. I don’t understand why this is there...

About VIDEO - discussion
http://gallery.ru-board.com/topic.cgi?forum=5&topic=49744&start=3700#18

1. All movies
https://ok.ru/video/90667747991
https://ok.ru/video/341008452247
https://ok.ru/video/363358653079
include 1-2 versions of HD; quality> = 720p.
It’s just with this quality that my reproduction begins.

These videos are played only on the line of browsers (I'm talking about the newest Pale Moon for XP fork) only on BOC/UXP for XP (Borealis, Bnavigator) from rtfreesoft.blogspot.com (github.com/roytam1/UXP/).

I reviewed 3 releases of each of these assemblies:
Mypal (mypal-28.8.4, mypal-28.8.3, mypal-28.8.3_MinGW_VS2010) and all modifications of the "factory" rtfreesoft.blogspot.com (BOC/UXP for XP (Borealis, Bnavigator), IceApe-UXP for XP, Pale Moon for XP XP ( NM28), Serpent/UXP for XP (Basilisk for XP).

The settings about:config BOC/UXP for XP and Serpent/UXP for XP are identical (they differ from the settings of other browsers not only in terms of WebRTC).

They give a similar result:

1. ECMAScript Language test262
https://v8.github.io/test262/website/default.html#

2. HTML5test
https://html5test.com/

However, BOC / UXP plays this VIDEO (with or without a flash plugin), while Serpent / UXP does not. 
The code for these browsers is very similar. Perhaps it is possible to find differences and the source of the bug, which also appeared in other browsers, including Mypal?

But this is such an ... interesting theoretical question.
I am a fan of the ideas you expressed here:
https://github.com/Feodor2/Mypal27old/wiki/H264-support

I use BSPlayer (LAV Filter 0.70.2 + FFDShow).
I'm sending a VIDEO from the browser into the player with the extension yt2p-2.1.1.xpi directly. Or through youtube-dl (which connects to yt2p-2.1.1 as a player).
Youtube-dl also works with Odnoklassniki. And not only ;)
http://ytdl-org.github.io/youtube-dl/supportedsites.html

The site addons.mozilla.org even saved the README of this (pre-Quantum!) version yt2p-2.1.1., unlike the developer.
https://addons.mozilla.org/en-US/firefox/addon/yt2p/
https://github.com/sumzary/yt2p/releases/tag/v2.1.1

**Comment by eugen1962** [comment]
> 
> 
> Why again? It was always like this, and comes from the mozilla, why this practice is going is another question. I want to get the actually reason for consider.
> 
> > The service pack installer again creates a section and registry entries
> 
> Where did you get it, i publish one regular installer only it is not to touch registry in the portable mode.
> 
> I can do an option in the installer maybe, if copying dlls is critically annoying for another one.
> 
> And i shall consider about webgl settings

Of course, the installer bug is inherited from Mozilla.
Of course, it's not about the portable version of Mypal.
Nothing is critical, I was simply annoyed, and did not express a claim.

About WebGL based on your comment (mypal-28.8.3_MinGW_VS2010).
I think that the driver driver 8.970.100.0 (07/01/2014) that I found and installed for the ATI Radeon HD 3450 required some registered library from MSVS. Therefore, I did not work with WebGL before installing the package and registering the libraries.
About the availability of WebGL2 and Direct3D11 in the browser code for WindowsXP. I don’t understand why this is there...

About VIDEO - discussion
http://gallery.ru-board.com/topic.cgi?forum=5&topic=49744&start=3700#18

All movies:
https://ok.ru/video/90667747991
https://ok.ru/video/341008452247
https://ok.ru/video/363358653079
include 1-2 versions of HD; quality> = 720p.
It’s just with this quality that my reproduction begins.

These videos are played (I'm talking about the newest Pale Moon for XP fork's) only on BOC/UXP for XP (Borealis, Bnavigator) from rtfreesoft.blogspot.com (github.com/roytam1/UXP/).

I reviewed 3 releases from each project:
Mypal (mypal-28.8.4, mypal-28.8.3, mypal-28.8.3_MinGW_VS2010) and all modifications of the "factory" rtfreesoft.blogspot.com (BOC/UXP for XP (Borealis, Bnavigator), IceApe-UXP for XP, Pale Moon for XP XP ( NM28), Serpent/UXP for XP (Basilisk for XP).

The settings about:config BOC/UXP for XP and Serpent/UXP for XP are identical (they differ from the settings of other browsers not only in terms of WebRTC).

They give a similar result:

    ECMAScript Language test262
    https://v8.github.io/test262/website/default.html#

    HTML5test
    https://html5test.com/

However, BOC / UXP plays this VIDEO (with or without a flash plugin), while Serpent / UXP does not. The code for these browsers is very similar. Perhaps it is possible to find differences and the source of the bug, which also appeared in other browsers, including Mypal?

But this is such an ... interesting theoretical question.
I am a fan of the ideas you expressed here:
https://github.com/Feodor2/Mypal27old/wiki/H264-support

I use BSPlayer (LAV Filter 0.70.2 + FFDShow).
I'm sending a VIDEO from the browser into the player with the extension yt2p-2.1.1.xpi directly. Or through youtube-dl (which connects to yt2p-2.1.1 as a player).
Youtube-dl also works with Odnoklassniki. And not only ;)
http://ytdl-org.github.io/youtube-dl/supportedsites.html

The site addons.mozilla.org even saved the README of this (pre-Quantum!) version yt2p-2.1.1., unlike the developer.
https://addons.mozilla.org/en-US/firefox/addon/yt2p/
https://github.com/sumzary/yt2p/releases/tag/v2.1.1

**Comment by letmeindude** [comment]
> 
> I use BSPlayer (LAV Filter 0.70.2 + FFDShow).
> I'm sending a VIDEO from the browser into the player with the extension yt2p-2.1.1.xpi directly. Or through youtube-dl (which connects to yt2p-2.1.1 as a player).
> Youtube-dl also works with Odnoklassniki. And not only ;)
> http://ytdl-org.github.io/youtube-dl/supportedsites.html
> 
Hi eugen1962,
How do you use yt2p-2.1.1 on ok.ru when yt2p-2.1.1 only works on youtube links?

**Comment by eugen1962** [comment]
> 
> 
> > I use BSPlayer (LAV Filter 0.70.2 + FFDShow).
> > I'm sending a VIDEO from the browser into the player with the extension yt2p-2.1.1.xpi directly. Or through youtube-dl (which connects to yt2p-2.1.1 as a player).
> > Youtube-dl also works with Odnoklassniki. And not only ;)
> > http://ytdl-org.github.io/youtube-dl/supportedsites.html
> 
> Hi eugen1962, how do you use yt2p-2.1.1 on ok.ru, when yt2p-2.1.1 works only on youtube links?

I wrote that Youtube-dl works with Odnoklassniki, and not yt2p-2.1.1.xpi:

Youtube-dl also works with Odnoklassniki. And not only;)
http://ytdl-org.github.io/youtube-dl/supportedsites.html

I did not write that yt2p-2.1.1.xpi supports Odnoklassniki.

Is it possible to expand the functionality of yt2p-2.1.1 so that it can support other sites? I think so. If you add algorithms for VIDEO (Odnoklassniki and other non-YouTube sites), taking them in the same Youtube-dl.

**Comment by letmeindude** [comment]
I would recommend Open With 6.x.x, because it is not limited to YouTube.
https://github.com/darktrojan/openwith/releases

**Comment by eugen1962** [comment]
> 
> 
> I would recommend Open With 6.x.x, because it is not limited to YouTube.
> https://github.com/darktrojan/openwith/releases

Thank you!

**Comment by Feodor2** [comment]
Well let we keep order here, one issue - one topic.
I am looking the good explanation why mozzila make duplicate of system dlls in the browser folder

**Comment by eugen1962** [comment]
> 
> 
> Well let we keep order here, one issue - one topic.
> I am looking the good explanation why mozzila make duplicate of system dlls in the browser folder

OK.
Looking at the shortened dependentlibs.list in an alternative project (UXP for XP), you see that they paid attention to this too.
By the way, it is very interesting to find out why in your project, unlike the alternative, there is no separate library mozsqlite3.dll. This is a really noticeable distinguishing feature of your development. How did it happen?

**Comment by Feodor2** [comment]
This is introduced in 28.9
https://github.com/Feodor2/Mypal/commit/2711cada39c0b5f455de5728f29c570ea6fcde5a
Others most likely will have this too

**Comment by eugen1962** [comment]
> 
> 
> This is introduced in 28.9
> [2711cad](https://github.com/Feodor2/Mypal/commit/2711cada39c0b5f455de5728f29c570ea6fcde5a)
> Others most likely will have this too

Thanks for the clarification and link!

**Comment by Feodor2** [comment]
I shall look into the installer stuff soon on building next version

**Comment by eugen1962** [comment]
> 
> 
> I shall look into the installer stuff soon on building next version

Thank you for your attention to users! 
The release of 28.9.0.02 works faster than 28.8.4. Thanks!
Will there be a package "mingw build # 126 without api_ms_win_core", as you called it, for new releases? Not a problem, if not, but I was interested in testing and comparing both versions of 28.8.3. 
But the difficult VIDEO (OK.RU) on 28.9.02 does not play either :( 
If this is not the code of the browser itself, then only the format of the audio or video track that internal codecs don't read is the reason for this. The guy who compiles Borealis - BOC / UXP solved this issue and chuckles at the others

**Comment by Feodor2** [comment]
I searched and did not find where registry entries used, and removed them from installer.

**Comment by eugen1962** [comment]
> 
> 
> I searched and did not find where registry entries used, and removed them from installer.

Thanks for resolving this issue! I read your comment and now deleted the registry entries and subfolders

HKEY_LOCAL_MACHINE \ SOFTWARE \ Mypal 28.8.4 \ extensions

Components REG_SZ C:\Intel\Mypal\components
Plugins    REG_SZ C:\Intel\Mypal\plugins

C:\Intel\Mypal\components
C:\Intel\Mypal\plugins

**If you find time, please help deal with WebGL**
HW acceleration and WebGL1 significantly accelerated the browser, especially 28.9.
But I was able to start fully WebGL1 + WebGL2 exclusively by disabling ANGLE, according to one of Mozilla's recommendations (Backend 2):

Platform/GFX/WebGL/Backends
https://wiki.mozilla.org/Platform/GFX/WebGL/Backends

Expected Platform/Layers/WebGL Combinations

Platform 	Windows XP
**Backend 1** (Compositing Accel): D3D9, ANGLE EGL+GLES2
**Backend 2** (Compositing Accel): D3D9, WGL, Native GL

webgl.disable-angle default boolean false >>> webgl.disable-angle user set boolean true
layers.acceleration.force default boolean false >>> layers.acceleration.force user set boolean true

But this is not a goog thing, ANGLE works better, faster, fewer errors. 
I do not understand the cause of the conflict...

1. Video card
2. Video card driver
3. Settings about: config
4. features of moz.build / source Mypal (for Windows XP)

I did not find a description of WebGL problems on the network due to the fault of the video card and driver 8.970.100.0 (problems with Catalyst 12, but not 14.4 / XP are described)

About: config settings I learn in WebGL Conformance Test Version 1.0.3 (WebGL Conformance Tests. Version> = 1.0.4 the browser partially passes, then the tests correctly ask you to shut down, as they will access the memory area reserved for higher versions of Windows)
https://www.khronos.org/registry/webgl/conformance-suites/1.0.3/webgl-conformance-tests.html

webgl.enable-debug-renderer-info >>> user set boolean true

**ANGLE + D3D9**
User Agent: Mozilla/5.0 (Windows NT 5.1; rv:68.9) Gecko/20100101 Goanna/4.5 Firefox/68.9 Mypal/28.9.0
WebGL VENDOR: Mozilla
WebGL VERSION: **WebGL 1.0**
WebGL RENDERER: Mozilla
Unmasked VENDOR:  ATI Technologies Inc.
Unmasked RENDERER: ATI Radeon HD 3450
WebGL R/G/B/A/Depth/Stencil bits (default config): 8/8/8/8/16/0

Test Summary (27899 total tests):
Tests PASSED: 27865
Tests FAILED: **34**
Tests TIMED OUT: 2
Tests SKIPPED: 0

**Backend 2 (WGL, Native GL) + D3D9**
User Agent: Mozilla/5.0 (Windows NT 5.1; rv:68.9) Gecko/20100101 Goanna/4.5 Firefox/68.9 Mypal/28.9.0
WebGL VENDOR: Mozilla
WebGL VERSION: **WebGL 2.0**
WebGL RENDERER: Mozilla
Unmasked VENDOR: ATI Technologies Inc.
Unmasked RENDERER: ATI Radeon HD 3450
WebGL R/G/B/A/Depth/Stencil bits (default config): 8/8/8/8/24/0

Test Summary (33960 total tests):
Tests PASSED: 33718
Tests FAILED: **242**
Tests TIMED OUT: 7
Tests SKIPPED: 0
....................... etc.

There will be a very long text if I describe the details. 
I saw your comment about GFX (skia, direct2d1 etc.) and understand the problem of lack of time:

_Problem with rendering some fonts in MyPal #17_
_Enersil opened this issue on 15 Oct 2018
https://github.com/Feodor2/Mypal/issues/17

Feodor2 commented on 18 Oct 2018
https://github.com/Feodor2/Mypal/issues/17#issuecomment-430817108_

If I distract you too much, no problem, just say it and we will stop the discussion

![about_support_ANGLE_1](https://user-images.githubusercontent.com/17965846/79268456-7b341000-7ea3-11ea-83ed-51027b909cec.jpg)
![about_support_ATI_1](https://user-images.githubusercontent.com/17965846/79268499-87b86880-7ea3-11ea-96c3-da52f8df893f.jpg)
![GPU1](https://user-images.githubusercontent.com/17965846/79268515-8f780d00-7ea3-11ea-89b3-afeef2cfe276.gif)
![GPU2](https://user-images.githubusercontent.com/17965846/79268535-9868de80-7ea3-11ea-9ed7-405065b71d5f.gif)
![GPU3](https://user-images.githubusercontent.com/17965846/79268548-9dc62900-7ea3-11ea-9736-5d4343154e35.gif)

**Comment by Feodor2** [comment]
So installer problem is solved.
Regarding webgl and other such let we discuss on their corresponding topics.
As i recall new webgl does not work on winxp, it needed so few people and has lowest priority for me too, but require an effort and experience.

**Comment by eugen1962** [comment]
> 
> 
> So installer problem is solved.
> Regarding webgl and other such let we discuss on their corresponding topics.
> As i recall new webgl does not work on winxp, it needed so few people and has lowest priority for me too, but require an effort and experience.

Thanks for solving the installer problem. There was a problem with redundant copying of MS system dlls to the browser folder.
As for WebGL ... I do not insist on discussing this issue. Moreover, Mozilla now calls this, including WebGL2, obsolete technology and has taken a great interest in WebGPU... I do not agree with that. However, as with the fact that WebGL2 does not work in WindowsXP. Take a look at the screenshot - I launched WebGL2 without ANGLE through OpenGL. Without MESA, LLVM etc.
Well, all right, if for you this is a very low priority and these questions are of interest to a very small number of people, why write so much about it? No reason. 
By the way, thanks for the new release. I have already updated my browser. Everything flies :)
